app/services/piece-access-level-service.js

```diff
@@ -62,4 +66,32 @@ export default class PieceAccessLevelService extends Service {
       previousPiece = await previousPiece.previousPiece;
     }
   }
+
+  /*
+   * This method is used to update the access levels of the pieces of a subcase's treatments. When a subcase has
+   * been updated to being confidential, all the pieces connected to it through agenda-item-treatments get the
+   * "Ministerraad" access level.
+   */
+  async updateDecisionsAccessLevelOfSubcase(subcase) {
+    await subcase.preEditOrSaveCheck();
+
+    if (!subcase.hasDirtyAttributes) {
+      return;
+    }
+
+    let { confidential: [, confidential] = [] } = subcase.changedAttributes();
+    if (!confidential) {
+      return;
+    }
```

Comment:
I'm a bit worried about relying on dirty state in a service while saving the record is managed outside of the service. It seems prone to (silent) errors in the future to me. I.e. moving the save operation before calling this method will make the logic silently fail.
I would check the precondition (subcase becoming confidential) outside of this service.